tests/test_decorators.py

```diff
@@ -518,3 +518,33 @@ def validate_many(self, data, many):
         errors = schema.validate([{'foo': 3, 'bar': 'not an int'}], many=True)
         assert 'bar' in errors[0]
         assert '_schema' not in errors
+
+    def test_allow_reporting_field_errors_in_schema_validator(self):
+
+        class NestedSchema(Schema):
+            baz = fields.Int(required=True)
+
+        class MySchema(Schema):
+            foo = fields.Int(required=True)
+            bar = fields.Nested(NestedSchema, required=True)
+            bam = fields.Int(required=True)
+
+            @validates_schema(skip_on_field_errors=True)
+            def consistency_validation(self, data):
+                errors = {}
+                if data['bar']['baz'] != data['foo']:
+                    errors['bar'] = {'baz': 'Non-matching value'}
+
+                if data['bam'] > data['foo']:
+                    errors['bam'] = 'Value should be less than foo'
+
+                if errors:
+                    raise ValidationError(errors)
+
+        schema = MySchema()
+        errors = schema.validate({'foo': 2, 'bar': {'baz': 5}, 'bam': 6})
```

Comment:
> I don't understand why I need to persuade you to use this feature.

If this were purely a new feature it would not be an issue, but you had to [modify existing test cases](https://github.com/marshmallow-code/marshmallow/pull/442/files#diff-8107272ea49a7ea4505985b05542ce00R449), which makes this PR backwards incompatible.

You are arguing that the documentation suggests there is a feature you want, but it is not implemented, so it is a bug that needs to be fixed. I am arguing that your interpretation of the documentation conflicts with the documentation for the `message` argument, the `ValidationError` implementation, and the `decorator` tests. One ambiguous footnote is not a strong argument for introducing breaking changes in v2.

> If your use cases are fine without it, don't use it.

The test cases you changed represent a use case that is not "fine without it". Those test cases are supported by a reasonable interpretation of the existing documentation.

> I haven't seen a better suggestions to [report] multiple different errors for different (nested) fields from whole-schema validator

You could break your whole-schema validator up into several schema validators, so that you can raise one `ValidationError` per error message. If you want to be able to assign error messages deeply nested fields, I would be interested in reviewing a PR that adds the dotted nesting syntax to the `fields` argument which would be a backwards compatible change.